File: backend/internal/modules/addresses/domain/services/addressService.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/addresses/application/dto"
	"github.com/javierjpv/edenBooks/internal/modules/addresses/domain/entities"
)

type AddressService interface {
	CreateAddress(address dto.AddressRequest) (*entities.Address, error)

	UpdateAddress(id uint, a dto.AddressRequest) error

	DeleteAddress(id uint) error

	GetAddressByID(id uint) (*entities.Address, error)

	CheckExistingAdress(a dto.AddressRequest) (bool, error)

	GetFilteredAddresses(filters map[string]string) ([]entities.Address, error)
}

File: frontend/src/modules/addresses/services/addressService.ts
import { addressApi } from "../api/addressApi";
import { IAddressRequest } from "../interfaces/IAddressRequest";
import { IAddressService } from "../interfaces/IAddressService";

export const AddressService: IAddressService = {
  CreateAddress: async (address: IAddressRequest) =>addressApi.CreateAddress(address),
  GetAddresss: async () => addressApi.GetAddresss(),
  GetAddressById: async (id: number) => addressApi.GetAddressById(id),
  UpdateAddress: async (id: number, address: IAddressRequest) =>addressApi.UpdateAddress(id, address),
  DeleteAddress: async (id: number) => addressApi.DeleteAddress(id),
};
